Makefile

```diff
@@ -20,38 +20,42 @@ CUT			=	\033[K
 
 #source files
 SRC_FILES	=	main \
-				Util \
 				CgiHandler \
+				Client \
+				ConfigParser \
+				Connection \
+				ExceptionManager \
+				FileHandler \
 				HttpRequest \
-				HttpRequestParser \
 				HttpRequestHandler \
+				HttpRequestParser \
 				HttpResponse \
-				Socket \
-				Server \
-				ExceptionManager \
 				HttpResponseParser \
-				FileHandler \
-				ConfigParser \
-				Client \
-				Connection \
-				ServerHandler
+				Methods \
+				Server \
+				ServerHandler \
+				Socket \
+				Util
```

Comment:
Im surprised by the order. I assume that the parts that are used in other places most like Util, Exceptions etc need to be the first ones to get compiled and the ones that use a lot of other classes like requestparser and serverhandler are the last ones to get compiled.